### PR TITLE
Ability to rename pages in site editor without title block

### PR DESCRIPTION
**Pull request by SaxonF**
image

## What?
Adds ability to rename pages when in site editor when no post title block exists in template. This PR also adds rename menu item to templates as well to be consistent.

## Why?
https://github.com/WordPress/gutenberg/issues/52763

## How?
Updates existing rename menu item component for templates to be used for any post type.

## Testing Instructions
1. Open a page in the site editor
2. Open page inspector and click more menu -> rename
3. Edit name of page and notice it updates
4. Do the same in the details panel in site editor (left nav)

### CODE REVIEW

**Comment by jameskoster** [comment]
Nice. 

On small issue I'm seeing is an error if I try to rename to empty. 

Screenshot 2023-09-20 at 14 25 48

In the case of templates this is fine. 

I appreciate it may be an edge case, but since pages can be published without a title, should we support that here too?

File: packages/edit-site/src/components/rename-menu-item/index.js [hunk 1]
```diff
@@ -33,15 +30,15 @@ export default function RenameMenuItem( { template, onClose } ) {
 	const { createSuccessNotice, createErrorNotice } =
 		useDispatch( noticesStore );
 
-	if ( template.type === TEMPLATE_POST_TYPE && ! template.is_custom ) {
+	if ( item.type === 'wp_template' && ! item.is_custom ) {
```
Comment (scruffian) [inline packages/edit-site/src/components/rename-menu-item/index.js:33]: Why are we not using the `TEMPLATE_POST_TYPE` const anymore?

File: packages/edit-site/src/components/page-actions/trash-page-menu-item.js [hunk 2]
```diff
@@ -51,11 +51,7 @@ export default function TrashPageMenuItem( { postId, onRemove } ) {
 	}
 	return (
 		<>
-			<MenuItem
-				onClick={ () => removePage() }
-				isDestructive
-				variant="secondary"
-			>
+			<MenuItem onClick={ () => removePage() } isDestructive>
```
Comment (scruffian) [inline packages/edit-site/src/components/page-actions/trash-page-menu-item.js:54]: Is this change meant to be part of this PR? I didn't see it mentioned in the description...

File: packages/edit-site/src/components/page-actions/index.js [hunk 3]
```diff
@@ -9,18 +9,20 @@ import { moreVertical } from '@wordpress/icons';
  * Internal dependencies
  */
 import TrashPageMenuItem from './trash-page-menu-item';
+import EditPageMenuItem from '../rename-menu-item';
 
-export default function PageActions( { postId, toggleProps, onRemove } ) {
+export default function PageActions( { post, toggleProps, onRemove } ) {
```
Comment (scruffian) [inline packages/edit-site/src/components/page-actions/index.js:14]: Is there a reason to change this prop? If possible its always better to pass simpler props as it prevents unnecessary rerenders in the component.

File: packages/edit-site/src/components/sidebar-edit-mode/page-panels/index.js [hunk 4]
```diff
@@ -20,30 +20,26 @@ import { store as editSiteStore } from '../../../store';
...
+		return {
+			hasResolved: hasFinishedResolution(
+				'getEditedEntityRecord',
+				queryArgs
+			),
+			page: record,
+		};
+	}, [] );
+
+	const { id, type, status, date, password, title, modified } = page;
```
Comment (scruffian) [inline packages/edit-site/src/components/sidebar-edit-mode/page-panels/index.js:42]: nice! We just need to make sure that if page doesn't exist this doesn't throw any errors.

**Comment by jasmussen** [comment]
This is a nice PR, and adds a nice added affordance. @jameskoster we should be able to fit this into the new summary card, no?

Needs a rebase, but would be good to land! 🚀

**Comment by aaronrobertshaw** [comment]
It proved easiest (at least for me) to create a fresh PR off trunk for this. The alternative is up in https://github.com/WordPress/gutenberg/pull/60232 so I'll close this once for now.

**Pull request closed without merging**